src/bootstrap/util.rs

```diff
@@ -324,3 +325,86 @@ pub fn symlink_dir(src: &Path, dest: &Path) -> io::Result<()> {
         }
     }
 }
+
+/// An RAII structure that indicates all output until this instance is dropped
+/// is part of the same group. On Travis CI, these output will be folded by
+/// default. This reduces visibility of unnecessary logs, allowing developers to
+/// quickly identify the error.
+pub struct OutputFolder<D: Display> {
+    name: D,
+    start_time: SystemTime, // we need SystemTime to get the UNIX timestamp.
+}
+
+impl<D: Display> OutputFolder<D> {
+    /// Creates a new output folder with the given group name.
+    pub fn new(name: D) -> OutputFolder<D> {
+        print!("travis_fold:start:{0}\ntravis_time:start:{0}\r\x1b[0K", name);
```

Comment:
flush: We could flush, but it is unnecessary. Travis will not fold until the "end" is seen, so putting a flush at the "end" should be good enough.